src/components/home-container/index.jsx

```diff
@@ -45,10 +67,18 @@ function HomeContainer() {
           />
         )}
 
-        <ExistingTaskCardContaienr tasks={tasks.slice(0, numOfCardVisible)} />
+        {!isTaskEmpty && (
+          <ExistingTaskCardContaienr
+            tasks={filteredTasks.slice(0, numOfCardVisible)}
+          />
+        )}
```

Comment:
Markup should always be as business logic free as it can be

```suggestion
        const tasks = filteredTasks.slice(0, numOfCardVisible)
        
        {!isTaskEmpty && (
          <ExistingTaskCardContaienr
            tasks={tasks}
          />
        )}
```